src/vmm/src/vstate/memory.rs

```diff
@@ -25,6 +32,174 @@ pub type GuestMmapRegion = vm_memory::MmapRegion<Option<AtomicBitmap>>;
 
 const GUARD_PAGE_COUNT: usize = 1;
 
+/// Errors associated with dumping guest memory to file.
```

Comment:
Would it not be worth keeping it in a separate file? What do we gain from merging into one?

Reply:
I think keeping all memory related code in 1 file is better, considering there is not that much of it.